main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -33,7 +33,9 @@ object TypedAst {
                   typeAliases: Map[Symbol.TypeAliasSym, TypedAst.TypeAlias],
                   entryPoint: Option[Symbol.DefnSym],
                   sources: Map[Source, SourceLocation],
-                  classEnv: Map[Symbol.ClassSym, Ast.ClassContext])
+                  classEnv: Map[Symbol.ClassSym, Ast.ClassContext],
+                  defSyms: Map[List[String], Symbol.DefnSym]
```

Comment:
I would put it first, and into one data structure  (i.e. one data structure for both defs and enums and ...)

Reply:
I also would like some Ast.Namespace abstraction. (Perhaps we should do that first?)